### PR TITLE
Add pycutest

### PR DESCRIPTION
**Pull request by AntoineD**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by AntoineD** [comment]
@conda-forge/help-python 
Ready for review.

**Comment by AntoineD** [comment]
@conda-forge/help-python
Ready for review.

**Comment by AntoineD** [comment]
Hello @conda-forge/staged-recipes,
please let me know if there is a chance this PR will be accepted or if I can do something to get it acceptable.

File: recipes/pycutest/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - make
+  host:
+    - pip
+    - python
+  run:
+    - numpy >=1.11
+    - python
+    - scipy >=0.17
+    - gcc_linux-64
+    - gfortran_linux-64
```
Comment (xhochy) [inline recipes/pycutest/meta.yaml:47]: These shouldn't be in `run`.
```suggestion
```

File: recipes/pycutest/activate.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export MASTSIF=${share_path}/mastsif
+
+export MYARCH="pc64.lnx.gfo"
+
+# backup initial standard variables to restore them in deactivate.sh
+export _CONDA_PYCUTEST_OLD_PATH=${PATH}
+export _CONDA_PYCUTEST_OLD_MANPATH=${MANPATH}
+
+export PATH=${SIFDECODE}/bin:${PATH}
+export PATH=${CUTEST}/bin:${PATH}
```
Comment (xhochy) [inline recipes/pycutest/activate.sh:15]: Why aren't the binaries installed in  `${PREFIX}/bin`? I would rather not touch `$PATH`.

File: recipes/pycutest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    folder: archdefs
+  - url: https://github.com/ralna/SIFDecode/archive/refs/tags/{{ sifdecode_version }}.zip
+    sha256: cfc0a5a82caada92da29b97cc26603f660bf1ddde0478cabf14f5d4740a0f385
+    folder: sifdecode
+  - url: https://github.com/ralna/CUTEst/archive/refs/tags/{{ cutest_version }}.zip
+    sha256: 537f3ab5688ef92b102f8f23f2da125f1e798beeb808e261aec4dee02b4eb1aa
+    folder: cutest
+  - url: https://bitbucket.org/optrove/sif/get/{{ sif_version }}.zip
+    sha256: a23214f2fb285860caa27bf05a767d75a441574ab126d60f004f9318a5911b5b
+    folder: mastsif
```
Comment (xhochy) [inline recipes/pycutest/meta.yaml:26]: Do they have to be shipped with this package? Couldn't they live in their own packages instead.

**Review by xhochy — COMMENTED** [review]
I have no idea about what `pycutest` is doing but I put in some general remarks.

File: recipes/pycutest/meta.yaml (hunk 1 above)
Comment (AntoineD) [reply recipes/pycutest/meta.yaml:47]: Once installed, pycutest may compile some pieces of code, this is why the recipe installs compilers for the runtime.

Comment (xhochy) [reply recipes/pycutest/meta.yaml:47]: The you should put in here the `compiler` macros and not the rendered names.

File: recipes/pycutest/activate.sh (hunk 2 above)
Comment (AntoineD) [reply recipes/pycutest/activate.sh:15]: I tried to get pycutest to work by installing things in the standard locations, but unfortunately there are hardcoded paths that rely on the executable to be installed in `${SIFDECODE}/bin` and `${CUTEST}/bin`.

File: recipes/pycutest/meta.yaml (hunk 3 above)
Comment (AntoineD) [reply recipes/pycutest/meta.yaml:26]: May be, I do not know how usable are those packages alone.

**Comment by AntoineD** [comment]
Hi @xhochy, thanks for taking a look at this PR :-)

Pycutest is a package used for benchmarking optimization algorithms and is a pain to install and get working for scientific people. I'm not a user but my colleagues do.

File: recipes/pycutest/meta.yaml (hunk 1 above)
Comment (AntoineD) [reply recipes/pycutest/meta.yaml]: Sure, I have no idea why I didn't use the macros!

File: recipes/pycutest/meta.yaml (hunk 3 above)
Comment (AntoineD) [reply recipes/pycutest/meta.yaml:26]: Is it really required that they live in separated packages?

**Pull request closed without merging**